hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/TimesTenDialect.java

```diff
@@ -90,29 +98,56 @@ protected String columnType(int sqlTypeCode) {
 			//      for the default Oracle type mode
 			//      TypeMode=0
 			case SqlTypes.BOOLEAN:
+			case SqlTypes.BIT:
 			case SqlTypes.TINYINT:
-				return "tt_tinyint";
+				return "TT_TINYINT";
 			case SqlTypes.SMALLINT:
-				return "tt_smallint";
+				return "TT_SMALLINT";
 			case SqlTypes.INTEGER:
-				return "tt_integer";
+				return "TT_INTEGER";
 			case SqlTypes.BIGINT:
-				return "tt_bigint";
+				return "TT_BIGINT";
 			//note that 'binary_float'/'binary_double' might
 			//be better mappings for Java Float/Double
 
+			case SqlTypes.CHAR:
+				return "CHAR($l)";
+			case SqlTypes.VARCHAR:
+			case SqlTypes.LONGVARCHAR:
+				return "VARCHAR2($l)";
+
+			case SqlTypes.BINARY:
+				return "BINARY($l)";
+			case SqlTypes.VARBINARY:
+			case SqlTypes.LONGVARBINARY:
+				return "VARBINARY($l)";
+
```

Comment:
Only need to handle the varchar type.

```suggestion
			case SqlTypes.LONGVARCHAR:
				return "varchar2($l)";
```